examples/backend/src/ExampleAPI.hs

```diff
@@ -19,18 +21,29 @@ module ExampleAPI
 where
 
 import Data.Aeson (FromJSON, ToJSON)
-import Data.Text (Text)
+import Data.ByteString (ByteString)
+import Data.Maybe (fromMaybe)
+import Data.Text as Text (Text, append, pack, unpack)
+import Data.Text.Encoding (decodeUtf8)
 import Elm (ElmStreet (..))
 import Elm.Generic (Elm (..))
 import GHC.Generics (Generic)
 import Network.Wai (Application)
-import Servant (Get, JSON, Post, Proxy (..), ReqBody, Server, serve, type (:<|>) (..), type (:>))
+import Servant (FromHttpApiData (parseHeader, parseQueryParam), Get, Handler, Header, Header', JSON, Post, Proxy (..), ReqBody, Required, Server, serve, type (:<|>) (..), type (:>))
 
 type UserAPI =
-  "users" :> Get '[JSON] [User]
-    :<|> "albert" :> Get '[JSON] User
+  "simple" :> "request" :> SimpleRequests
     :<|> "signup" :> ReqBody '[JSON] User :> Post '[JSON] User
-    :<|> "sth" :> Post '[JSON] User
+    :<|> "headers" :> Headers
+
+type SimpleRequests =
+  "list" :> Get '[JSON] [User]
+    :<|> "customType" :> Get '[JSON] User
+
+type Headers =
+  "basic" :> Header "someHeader" Text :> Post '[JSON] Text
+    :<|> "multiple" :> Header "someHeader1" Text :> Header' '[Required] "someHeader2" Int :> Post '[JSON] Text
+    :<|> "customType" :> Header "sortBy" SortBy :> Post '[JSON] SortBy
```

Comment:
How about returning `Maybe SortBy` here, so that you can easily see if the header was read correctly?

Reply:
Do I need to test it with another type or still test it with a Maybe value?